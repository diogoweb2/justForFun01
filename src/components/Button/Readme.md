Button example:

```js
<div>
  <Button>primary</Button>
  <Button type="link">link</Button>
  <Button type="secondary">secondary</Button>
</div>
```
