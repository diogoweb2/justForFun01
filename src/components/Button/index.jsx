import React from "react";
import PropTypes from "prop-types";

const Button = ({ onClick, children, type }) => {
  let typeClass;

  switch (type.toLowerCase()) {
    case "link":
      typeClass = "";
      break;
    case "secondary":
      typeClass = "slds-button_neutral";
      break;

    default:
      typeClass = "slds-button_brand";
      break;
  }

  typeClass += " slds-button";

  return (
    <button onClick={onClick} className={typeClass}>
      {children}
    </button>
  );
};

Button.propTypes = {
  onClick: PropTypes.func.isRequired,

  children: PropTypes.node.isRequired,
  /**
   * values: primary, link, secondary
   */
  type: PropTypes.string
};

Button.defaultProps = {
  type: "primary"
};

export default Button;
