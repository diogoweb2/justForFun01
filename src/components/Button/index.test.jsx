import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Button from "./index";

let wrapper;
describe("Button", () => {
  const fn = jest.fn();

  it("should render", () => {
    wrapper = shallow(<Button>test</Button>);
    expect(wrapper).toHaveLength(1);
  });

  it("should trigger click", () => {
    wrapper = shallow(<Button onClick={fn} />);
    wrapper.find("button").simulate("click");
    expect(fn).toHaveBeenCalled();
  });

  it("should change styles", () => {
    wrapper = shallow(<Button type="secondary">OK</Button>);
    expect(wrapper.find(".slds-button_neutral")).toHaveLength(1);
  });

  it("should match snapshot - Link...", () => {
    wrapper = shallow(<Button>OK</Button>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("should match snapshot - secondary...", () => {
    wrapper = shallow(<Button type="secondary">OK</Button>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it("should match snapshot - primary...", () => {
    wrapper = shallow(<Button type="primary">OK</Button>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
