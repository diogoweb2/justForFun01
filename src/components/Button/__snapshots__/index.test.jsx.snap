// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should match snapshot - Link... 1`] = `
<button
  className="slds-button_brand slds-button"
>
  OK
</button>
`;

exports[`Button should match snapshot - primary... 1`] = `
<button
  className="slds-button_brand slds-button"
>
  OK
</button>
`;

exports[`Button should match snapshot - secondary... 1`] = `
<button
  className="slds-button_neutral slds-button"
>
  OK
</button>
`;
